source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -176,6 +186,10 @@ PostIoAction SslSocket::doHandshake() {
   int rc = SSL_do_handshake(ssl_);
   if (rc == 1) {
     ENVOY_CONN_LOG(debug, "handshake complete", callbacks_->connection());
+    if (state_ == SocketState::HandshakeInProgress) {
+      // No need to disable if no async engine in use.
+      file_event_->setEnabled(0);
```

Comment:
Is this needed? I think it's an error if the OpenSSL engine tries to make the fd readable again?

Reply:
removed